src/transforms/reduce/transform.rs

```diff
@@ -760,4 +765,139 @@ merge_strategies.bar = "concat"
         })
         .await;
     }
+
+    #[tokio::test]
+    async fn strategy_path_with_nested_fields() {
+        let reduce_config = toml::from_str::<ReduceConfig>(indoc!(
+            r#"
+            group_by = [ "id" ]
+
+            merge_strategies.id = "discard"
```

Comment:
Could we add here one more field with default strategy please? For instance,
`{message : { a : number: 2}}` and ``{message : { a : number: 3}}` should be reduced to `{message : { a : number: 5}}`, because numbers are summed up by default.

Reply:
👍 Done in https://github.com/vectordotdev/vector/pull/20800/commits/a349551cd01a48bfb17d3806ea25c23c4ef84a96